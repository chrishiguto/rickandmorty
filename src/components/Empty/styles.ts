import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium};
    display: flex;
    flex-direction: column;
    align-items: center;
    color: ${theme.colors.white};
  `}
`

export const Title = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxlarge};
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    margin-top: ${theme.spacings.medium};
    margin-bottom: ${theme.spacings.small};
  `}
`

export const NotFound = styled.img`
  ${({ theme }) => css`
    width: 100%;
    max-width: 50rem;
    margin-top: ${theme.spacings.medium};
  `}
`
