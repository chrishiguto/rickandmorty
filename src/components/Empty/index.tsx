import Button from 'components/Button'

import * as S from './styles'

export type EmptyProps = {
  onGoBack?: () => void
}

const Empty = ({ onGoBack }: EmptyProps) => (
  <S.Wrapper>
    <S.Title>Oops!</S.Title>
    <S.Description>
      {`Looks like this character doesn't exist... Mhmm, try again!`}
    </S.Description>
    {!!onGoBack && <Button onClick={onGoBack}>Go back</Button>}
    <S.NotFound src={'/img/404.png'} />
  </S.Wrapper>
)

export default Empty
