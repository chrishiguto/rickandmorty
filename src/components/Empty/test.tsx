import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

describe('<Empty />', () => {
  it('should render the empty message without go back button by default', () => {
    const { container } = renderWithTheme(<Empty />)

    expect(screen.getByRole('heading', { name: /oops!/i })).toBeInTheDocument()
    expect(
      screen.getByText(
        /looks like this character doesn't exist... Mhmm, try again!/i
      )
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the empty message with go back button', async () => {
    const mockFn = jest.fn()

    renderWithTheme(<Empty onGoBack={mockFn} />)

    const button = screen.getByRole('button', { name: /Go back/i })

    userEvent.click(button)

    await waitFor(() => {
      expect(mockFn).toBeCalledTimes(1)
    })
  })
})
