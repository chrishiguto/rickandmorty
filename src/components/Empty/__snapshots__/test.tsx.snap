// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<Empty /> should render the empty message without go back button by default 1`
] = `
.c0 {
  margin: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #ffffff;
}

.c1 {
  font-size: 2.8rem;
}

.c2 {
  font-size: 2.0rem;
  margin-top: 3.2rem;
  margin-bottom: 2.4rem;
}

.c3 {
  width: 100%;
  max-width: 50rem;
  margin-top: 3.2rem;
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    Oops!
  </h3>
  <p
    class="c2"
  >
    Looks like this character doesn't exist... Mhmm, try again!
  </p>
  <img
    alt="Rick Sanchez not found image"
    class="c3"
    src="/img/404.png"
  />
</div>
`
